### PR TITLE
Fix to work on Mac and update README.md and .gitignore

### PR DESCRIPTION
**Pull request by nightmareci**
I've tested all three supported types of builds against the changes in this pull request (building with `tcc` on Linux with Wine, building on Linux, and even building both Apple Silicon and Intel Mac builds), and tested that all types of builds run successfully.

I changed how the Linux and Mac build commands receive SDL2, because the previous way SDL2 was provided didn't work on Apple Silicon using Homebrew, but the change fixes that. And using `sdl2-config` is the "right way" to use SDL2 with manually-written compile commands, because it provides some stuff SDL2 needs beyond `-lSDL2`.

In the process of fixing it for Mac, [I created a CMake build script](https://github.com/nightmareci/dos-like-cmake) to help with development, as using CMake makes it easier to develop in many IDEs. Plus, it compiles the base `dos-like` library separately from the samples, so all of the samples compile faster.

### CODE REVIEW

**Comment by nightmareci** [comment]
GCC and Clang don't output nicely named binaries by default the way TCC does, you have to use `-o` with a filename to get that. Plus, using `.bin` makes it easy to have the `.gitignore` file keep the binaries from being committed; not using any extension would mean we'd have to have the name of each program in `.gitignore`, not so fun for maintenance.

**Comment by mattiasgustavsson** [comment]
That’s a good point, I guess they default to naming it a.out? Then it makes sense to use the -o switch in the build all scripts, but I think it is better to keep it out of the readme - it is not essential to have a working build, it just makes it nicer, but the less switches there are in the directions of the readme, the better. And I think it is better to use .out as extension and in the gitignore, as that is the compiler default

File: source/dos.h [hunk 1]
```diff
@@ -2817,13 +2825,17 @@ typedef struct timecaps_tag { UINT wPeriodMin; UINT wPeriodMax; } TIMECAPS, *PTI
...
 #define TML_IMPLEMENTATION
+#ifdef APP_WINDOWS
 #pragma warning( push )
 #pragma warning( disable: 4201 )
+#endif
 #define TML_MALLOC tml_mus_custom_malloc
 #define TML_REALLOC tml_mus_custom_realloc
 #define TML_FREE tml_mus_custom_free
 #include "libs/tml.h"
+#ifdef APP_WINDOWS
```
Comment (mattiasgustavsson) [inline source/dos.h:2836]:
```suggestion
#ifdef _WIN32
```

File: source/dos.h [hunk 2]
```diff
@@ -2817,13 +2825,17 @@ typedef struct timecaps_tag { UINT wPeriodMin; UINT wPeriodMax; } TIMECAPS, *PTI
 #include "libs/tsf.h"
 
 #define TML_IMPLEMENTATION
+#ifdef APP_WINDOWS
```
Comment (mattiasgustavsson) [inline source/dos.h:2828]:
```suggestion
#ifdef _WIN32
```

File: source/dos.h [hunk 3]
```diff
@@ -202,20 +202,28 @@ int mousey( void );
...
 #pragma warning( push )
 #pragma warning( disable: 4201 )
+#endif
 #include "libs/tml.h"
+#ifdef APP_WINDOWS
 #pragma warning( pop)
+#endif
 
 
+#ifdef APP_WINDOWS
```
Comment (mattiasgustavsson) [inline source/dos.h:215]:
```suggestion
#ifdef _WIN32
```

File: source/dos.h [hunk 4]
```diff
@@ -202,20 +202,28 @@ int mousey( void );
...
 #pragma warning( push )
 #pragma warning( disable: 4024 )
 #pragma warning( disable: 4047 )
 #pragma warning( disable: 4242 )
 #pragma warning( disable: 4244 )
 #pragma warning( disable: 4701 )
+#endif
 #include "libs/gif_load.h"
+#ifdef APP_WINDOWS
```
Comment (mattiasgustavsson) [inline source/dos.h:224]:
```suggestion
#ifdef _WIN32
```

File: source/dos.h [hunk 5]
```diff
@@ -202,20 +202,28 @@ int mousey( void );
 #include "libs/thread.h"
 #include "libs/tsf.h"
 
+#ifdef APP_WINDOWS
```
Comment (mattiasgustavsson) [inline source/dos.h:205]:
```suggestion
#ifdef _WIN32
```

File: source/dos.h [hunk 6]
```diff
@@ -202,20 +202,28 @@ int mousey( void );
...
 #pragma warning( push )
 #pragma warning( disable: 4201 )
+#endif
 #include "libs/tml.h"
+#ifdef APP_WINDOWS
```
Comment (mattiasgustavsson) [inline source/dos.h:210]:
```suggestion
#ifdef _WIN32
```

**Pull request merged**